src/std/misc/text.ss

```diff
@@ -13,3 +22,169 @@
+      stx
+      (with-syntax ((stx stx))
+        #'(format "~a" stx))))
+
+  (def (map-format stx)
+    (stx-map stx-format stx))
+
+  ;; custom eval to capture the proper variables
+  (def (eval** stx ctx)
+    (parameterize ((current-expander-phi (1+ (current-expander-phi))))
+      (let (stx (datum->syntax ctx stx))
+        (eval-syntax stx #t))))
+
+  (def ht (datum->syntax #f 'ht)) ;; hash table placeholder
+  (def al (datum->syntax #f 'al)) ;; alist placeholder
+  (def pl (datum->syntax #f 'pl)) ;; plist placeholder
+
+  (def (parse-quasistring stx ctx rt-symbol-proc output-proc (expansion-time-eval #f))
+    ;; rt-symbol-proc wraps sexp if sexp is a run time symbol
+    ;; output-proc is what we return from the macro
+    ;; if expansion-time-eval is #t, allow for ##{} template variables
+    (def (lp (port (open-input-string (stx-e stx)))
+             (c    (read-char port))
+             (ops  [])  ;; arguments of string-append
+             (str  "")) ;; container to concatenate strings
+      (cond
+       ((eq? c #\#)
+        (let (c+1 (read-char port))
+          (cond
+           ((eq? c+1 #\{) ;; run time template variables
+            (let ((sexp (read port))
+                  (c+2  (read-char port)))
+              (if (eq? c+2 #\})
+                (if (symbol? sexp)
+                  (lp port (read-char port) (append ops [str (rt-symbol-proc sexp)]) "")
+                  (lp port (read-char port) (append ops [str sexp]) ""))
+                (error qs-err-missing-close c+2))))
+           ((eq? c+1 #\#) ;; expansion time template variables
+            (if expansion-time-eval
+              (let (c+2 (read-char port))
+                (if (eq? c+2 #\{)
+                  (let* ((sexp     (read port))
+                         (sexp-str (format "~a" (eval** sexp ctx)))
+                         (c+3      (read-char port)))
+                    (if (eq? c+3 #\})
+                      ;; allow for nested templates
+                      (let* ((rest (read-all-as-string port))
+                             (tmp  (string-append sexp-str rest))
+                             (new-port (open-input-string tmp)))
+                        (close-input-port port)
+                        (lp new-port (read-char new-port) ops str))
+                      (error qs-err-missing-close c+3)))
+                  (lp port (read-char port) ops (string-append str (list->string [c c+1 c+2])))))
+              (lp port (read-char port) ops (string-append str (list->string [c c+1]))))))))
+       ((eq? c #!eof)
+        (close-input-port port)
+        (with-syntax (((ops ...) (datum->syntax ctx (map-format (append ops [str])))))
+          (output-proc #'(ops ...))))
+       (else
+        (lp port (read-char port) ops (string-append str (list->string [c]))))))
+    (lp)))
+
+(defsyntax (defquasiparser stx)
+  (syntax-case stx ()
+    ((macro id rt-symbol-proc output-proc expand-time-eval)
+     (with-syntax ((id #'id)
```

Comment:
no need to templetaize `id` with `(id #'id)`, it's already a syntax template variable!

Reply:
Fixed.